src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -2562,10 +2552,32 @@ FixedwingPositionControl::get_nav_speed_2d(const Vector2f &ground_speed)
 	return nav_speed_2d;
 }
 
+float FixedwingPositionControl::compensateTrimThrottleForDensityAndWeight(float throttle_trim, float throttle_min,
+		float throttle_max)
+{
+	float weight_ratio = 1.0f;
+
+	if (_param_weight_base.get() > FLT_EPSILON && _param_weight_gross.get() > FLT_EPSILON) {
+		weight_ratio = math::constrain(_param_weight_gross.get() / _param_weight_base.get(), MIN_WEIGHT_RATIO,
+					       MAX_WEIGHT_RATIO);
+	}
+
+	float air_density_throttle_scale = 1.0f;
+
+	if (PX4_ISFINITE(_air_density)) {
+		// scale throttle as a function of sqrt(rho0/rho)
+		const float eas2tas = sqrtf(CONSTANTS_AIR_DENSITY_SEA_LEVEL_15C / _air_density);
+		air_density_throttle_scale = constrain(eas2tas, 1.f, 2.f);
```

Comment:
a bit pedantic - but should we have an "altitude ceiling" defined (and commented) here? otherwise seems like an arbitrary choice of maximum density scale. e.g. 2.0 here equates to something close to entering the stratosphere (11km) on the standard atmosphere table :) -- just like to be intentional on numbers